yarn-project/simulator/src/avm/avm_simulator.ts

```diff
@@ -83,10 +83,24 @@ export class AvmSimulator {
   public async execute(): Promise<AvmContractCallResult> {
     const bytecode = await this.context.persistableState.getBytecode(this.context.environment.address);
 
-    // This assumes that we will not be able to send messages to accounts without code
-    // Pending classes and instances impl details
     if (!bytecode) {
-      throw new NoBytecodeForContractError(this.context.environment.address);
```

Comment:
If we "throw" then we'd need to wrap calls to the AVM in try-catch. It would be fine for external calls, but breaks things for enqueued calls.

Reply:
Yes because this isn't happening inside the execution loop that has a try-catch. This is the very first few lines of `simulator.execute()`. If this throws, then the Call instruction needs to wrap calls to `simulator.execute()` in a try-catch. And external calls to that function (for enqueued calls) would need to handle the exception too.